pep/app/pods/components/document/text/component.ts

```diff
@@ -103,40 +245,96 @@ export default class DocumentText extends Component<DocumentTextArgs> {
     async viewGlossaryTerm(term: string, id?: string) {
         try {
             this.loadingBar.show();
+            const parsedTerm = term.split(';');
             const results = await this.store.query('glossary-term', {
                 termidtype: id ? 'ID' : 'Name',
-                termIdentifier: id ? id : term
+                termIdentifier: id ? id : parsedTerm[0]
             });
-            this.args.onGlossaryItemClick(term, results.toArray());
+            this.args.onGlossaryItemClick(parsedTerm[0], [...new Set(results.toArray())]);
         } catch (error) {
             throw error;
         } finally {
             this.loadingBar.hide();
         }
     }
 
+    /**
+     * Parse the document on change of the text
+     *
+     * @memberof DocumentText
+     */
     @action
     parseDocument() {
-        this.parseDocumentText(this.args.text);
+        this.parseDocumentText(
+            this.args.document.accessLimited ? this.args.document.abstractCleaned : this.args.document.documentCleaned
+        );
     }
 
+    /**
+     * set up the tooltips and if we get passed in a page number - scroll to it
+     *
+     * @param {HTMLElement} element
+     * @memberof DocumentText
+     */
     @action
     setupListeners(element: HTMLElement) {
         this.containerElement = element;
         scheduleOnce('afterRender', this, this.attachTooltips);
+        if (this.args.page) {
+            this.scrollToPage(parseInt(this.args.page, 10));
```

Comment:
use `Number(val)` instead of `parseInt(val, 10)` unless `val` is known to contain some garbage/non-number characters that you need to filter out. `parseInt()` handles that with more complex parsing, but in most cases you only need the simple casting/coercing that `Number()` provides.

Reply:
Yea, there is garbage in the values that needs to be converted.